### DISCUSSION TITLE
Reuse location data from other apps

### DISCUSSION THREAD

**Discussion by SKrywinski** [owntracks/android#1164] [Q&A]
Hi!
Thank you for this great piece of software!
I have a question referring to the [booklet](https://owntracks.org/booklet/features/location/#android), where it says that "Location data from other apps is reused and published as soon as it arrives. ". I know that there is another Issue asking for the ability to change location mode by movement detection and that there is the opportunity to do this by automation apps.
But does Owntracks reuse location data from Android as exact as it gets the data from another app? When I have Owntracks in "Significant location change mode" and I use another app for navigation, so my Android gets more precise location data, Owntracks could reuse this data to get more tracking points? (You can observe that with Google's location history: while navigating with no matter what app, it is more accurate than while not navigating.) In Owntracks (or better recorder) there are not more tracking points when I use navigation -  there are just as little points as I am expecting when not navigating in significant location change mode. 
Can someone give some clarification?

Thank you very much!

* **App build number**: 2.4.6 (24600) - OSS
* **Android version**: 11
* **Device**: Motorola Moto G30

**Comment by growse** [comment]
I think that part of the booklet is referring to the fact that on the Google Play version in significant changes mode, the app uses the Google Play Services `FusedLocationProviderClient` (https://developers.google.com/android/reference/com/google/android/gms/location/FusedLocationProviderClient) in a mode which allows it to opporunistically benefit from other applications that are listening to the GPS. Without other apps open, OT relies on the Google GSM tower / Wifi BSSID / A-GPS locator services to provide it with a periodic approxomation - it doesn't listen for GPS changes specifically.

On the OSS version, Google Play Services is not used, so the app falls back to using infrequent GPS, network and passive location receivers. Network and passive both rely on the device providing a way to map the network and GSM cell Id / WiFi BSSIDs to an approximate location, which may or may not be available depending on the Android build / flavour you have on your device.

Once the OSS build is actually ready for mainstream usage, we'll get the booklet updated.
